pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -140,12 +142,30 @@ func (a *AgentConfig) Load(f asset.FileFetcher) (bool, error) {
 		return false, errors.Wrapf(err, "failed to unmarshal %s", agentConfigFilename)
 	}
 
+	if len(config.Hosts) == 0 {
+		// Use baremetal hosts if defined in install-config
+		installConfig := &agentAsset.OptionalInstallConfig{}
+		found, err := installConfig.Load(f)
```

Comment:
This approach doesn't fit the asset framework IMHO, and it may be difficult to maintain: a single asset it's responsible for its own stuff.
Instead, such kind of multiple dependencies could be easily managed into another (synthetic) asset (like AgentManifests) that depends on both (AgentConfig and InstallConfig) and applies the required logic by sneaking in each asset artifacts,

Reply:
Updated to add AgentHosts.